<?xml version="1.0" ?>

<robot name="iCub" xmlns:xacro="http://ros.org/wiki/xacro">

  <group name="left_leg">
    <chain base_link="base_link" tip_link="l_sole"/>
  </group>
  <group name="right_leg">
    <chain base_link="base_link" tip_link="r_sole"/>
  </group>
  <group name="torso">
    <chain base_link="base_link" tip_link="torso"/>
  </group>
  <group name="left_arm">
    <chain base_link="chest" tip_link="l_gripper"/>
  </group>
  <group name="right_arm">
    <chain base_link="chest" tip_link="r_gripper"/>
  </group>
  <group name="neck">
    <chain base_link="chest" tip_link="head"/>
  </group>
  
  <group name="arms">
    <group name="left_arm"/>
    <group name="right_arm"/>
  </group>

  <group name="legs">
    <group name="left_leg"/>
    <group name="right_leg"/>
  </group>
  
   

  <group name="imu_sensors">
    <link name="imu_frame"/>
  </group>

  <!--<group name="force_torque_sensors">
    <joint name="r_leg_ft_joint"/>
    <joint name="l_leg_ft_joint"/>
  </group>-->

<group name="chains">
    <group name="left_leg"/>
    <group name="right_leg"/>
    <group name="torso"/>
    <group name="left_arm"/>
    <group name="right_arm"/>
    <group name="neck"/>
  </group>
  
  <group_state name="home" group="chains">
	<joint name="l_ankle_pitch" value="0.00"/>
	<joint name="l_ankle_roll" value="0.00"/>
        <joint name="l_elbow" value="0.97"/>
	<joint name="l_hip_pitch" value="-0.00"/>
	<joint name="l_hip_roll" value="-0.00"/>
        <joint name="l_hip_yaw" value="0.00"/>
	<joint name="l_knee" value="-0.00"/>
	<joint name="l_shoulder_pitch" value="-0.00"/>
        <joint name="l_shoulder_roll" value="0.00"/>
	<joint name="l_shoulder_yaw" value="-0.00"/>
	<joint name="l_wrist_pitch" value="-0.00"/>
	<joint name="l_wrist_prosup" value="0.00"/>
	<joint name="l_wrist_yaw" value="0.00"/>
        <joint name="neck_pitch" value="-0.00"/>
	<joint name="neck_roll" value="0.00"/>
	<joint name="neck_yaw" value="0.00"/>
        <joint name="r_ankle_pitch" value="-0.00"/>
	<joint name="r_ankle_roll" value="0.00"/>
	<joint name="r_elbow" value="0.97"/>
        <joint name="r_hip_pitch" value="-0.00"/>
	<joint name="r_hip_roll" value="-0.00"/>
	<joint name="r_hip_yaw" value="0.00"/>
	<joint name="r_knee" value="-0.00"/>
        <joint name="r_shoulder_pitch" value="-0.00"/>
	<joint name="r_shoulder_roll" value="0.00"/>
	<joint name="r_shoulder_yaw" value="-0.00"/>
        <joint name="r_wrist_pitch" value="-0.00"/>
	<joint name="r_wrist_prosup" value="0.00"/>
	<joint name="r_wrist_yaw" value="0.00"/>
        <joint name="torso_pitch" value="-0.00"/>
	<joint name="torso_roll" value="0.00"/>
        <joint name="torso_yaw" value="0.00"/>   
    </group_state>
</robot>
